tests/loggen/logline_generator.c

```diff
@@ -110,16 +111,29 @@ generate_log_line(char *buffer, int buffer_length, int syslog_proto, int thread_
 
   /* create time stamps */
   struct timeval now;
-  gettimeofday(&now, NULL);
+  static struct timeval ts_formatted = {0};
   struct tm tm;
-  char stamp[32];
-  localtime_r(&now.tv_sec, &tm);
-  int len = strftime(stamp, sizeof(stamp), "%Y-%m-%dT%H:%M:%S", &tm);
-  memcpy(&buffer[pos_timestamp2], stamp, len);
+  static char stamp[32] = {0};
+  static int len;
```

Comment:
These globals are written by multiple threads, the race can produce invalid timestamps.

We could use TLS here, or just cache the timestamp on the caller side.

Reply:
Good catch. I was concentrating so much on my single-threaded case...

Anyway, I just used a mutex here, generating that timestamp should not happen very frequently now, a mutex is just fine.